pkg/restapi/vc/operation/operations.go

```diff
@@ -990,16 +999,29 @@ func (o *Operation) parseAndVerifyVP(vpBytes []byte) (*verifiable.Presentation,
 }
 
 func getPublicKeyID(didDoc *did.Doc) (string, error) {
-	var publicKeyID string
-
 	switch {
 	case len(didDoc.PublicKey) > 0:
-		publicKeyID = didDoc.PublicKey[0].ID
+		var publicKeyID string
+
+		for _, k := range didDoc.PublicKey {
+			if strings.HasPrefix(k.Type, Ed25519VerificationKey) {
+				publicKeyID = k.ID
+				break
+			}
+		}
+
+		if !isDID(publicKeyID) {
```

Comment:
So their DID keys are not URIs (not as per spec). Can you add comment (or create issue) to remove this condition when they become compliant?

Reply:
DID element has existing issue to support public keys in with DID based ids. Created #140 
 and added a TODO to remove it in future.

Reply:
They should be compliant now: https://github.com/decentralized-identity/element/pull/179

https://uniresolver.io/1.0/identifiers/did:elem:EiBJJPdo-ONF0jxqt8mZYEj9Z7FbdC87m2xvN0_HAbcoEg